File: Minimum_depth_of_binary_tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int minDepth(TreeNode* root) {
        if(!root)   return 0;
        int level = 0;
        
        queue<TreeNode*> q;
        q.push(root);
        
        while(q.size())
        {
            int num = q.size();
            level++;
            while(num--)
            {
                TreeNode *cur = q.front();
                q.pop();
                if(cur->left) q.push(cur->left);
                if(cur->right) q.push(cur->right);
                if((cur->left == NULL) && (cur->right == NULL)) return level;
            }
        }
    }
};

File: leetcode/divideConquer&Tree/Minimum Depth of Binary Tree.go
package divideConquer_Tree

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func minDepth(root *TreeNode) int {
	if root == nil{
		return 0
	}
	if root.Left == nil && root.Right == nil{
		return 1
	}else if root.Left == nil{
		return 1 + minDepth(root.Right)
	}else if root.Right == nil{
		return 1 + minDepth(root.Left)
	}else{
		return 1 + min(minDepth(root.Left), minDepth(root.Right))
	}
}

func min(a, b int) int{
	if a < b{
		return a
	}
	return b
}
